### PR TITLE
OIDC Callback Error Redirect

### PR DESCRIPTION
**Pull request by talanknight**

### CODE REVIEW

File: internal/servers/controller/handlers/marshler.go [hunk 1]
```diff
@@ -0,0 +1,19 @@
+package handlers
+
+import (
+	"github.com/grpc-ecosystem/grpc-gateway/v2/runtime"
+	"google.golang.org/protobuf/encoding/protojson"
+)
+
+var JSONMarshaler = &runtime.JSONPb{
```
Comment (jimlambrt) [inline internal/servers/controller/handlers/marshler.go:8]: why not just provide a function that returns the needed runtime.JSONPb?  Is there some reason all of them need to share the same instance?

**Review by jimlambrt — CHANGES_REQUESTED** [review]
looks good.  just a few suggestions.

File: internal/errors/code.go [hunk 2]
```diff
@@ -85,4 +85,7 @@ const (
 
 	// External system errors are reserved codes 3000-3999
 	Unavailable Code = 3000 // Unavailable represents that an external system is unavailable
+
+	// OIDC authentication provided errors
+	OidcProviderCallbackError Code = 4000
```
[the remark this answers is not in the document]
Comment (talanknight) [reply internal/errors/code.go]: Thank you!

File: internal/servers/controller/handlers/marshler.go (hunk 1 above)
Comment (talanknight) [reply internal/servers/controller/handlers/marshler.go]: Thank you!

**Review by jimlambrt — APPROVED** [review]

**Pull request merged**